### DISCUSSION TITLE
本地部屬pip install遇到問題，求解(已解決)

### DISCUSSION THREAD

**Discussion by Hong60229** [svc-develop-team/so-vits-svc#309] [Q&A]
小弟我在本地部屬時於pip install pyworld praat-parselmouth fairseq librosa matplotlib tensorboard tensorboardX這一步時出現如圖片中的狀況。網上搜不到有遇到類似狀況的人，chatgpt回的解決方案也沒用，反覆嘗試依然失敗。
由於小弟我只在b站找到4.0的本地部屬教學，不清楚4.1是否適用，如果大佬們有更好的教學來源可以提供的話，我會很感謝。

7/22更新:有關圖片中出現的一段錯誤(C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.36.32532\include\yvals.h(17): fatal error C1083: 無法開?包含?案: 'crtdbg.h': No such file or directory)，我在網上搜到的多數解決方法都是清除舊版sdk再重裝vs，我試了許多次依然沒用。直到看到一篇文是說sdk需跟windos版本相對應(不只是7、10跟11，包括22H1跟22H2等等)，由於小弟我的電腦是win10 22H2，而win10sdk都是支援22H1以前的，所以我之前改成裝win10sdk才會沒用，稍早更新成win11後立刻就成功了!

![螢幕擷取畫面 2023-07-22 022743](https://github.com/svc-develop-team/so-vits-svc/assets/113456718/ef82686c-7292-4bef-9325-f17639acbc07)

**Comment by zxwwxz001** [comment]
我在安装pyworld的时候遇到了同样的错误，我像下面这样加上--no-build-isolation就没问题了
pip install pyworld --no-build-isolation
你加上--no-build-isolation试试，但是依赖包要先自己装。

**Reply by Hong60229** [reply]
謝謝回復，小弟是在fairseq那裡出了問題，照你說的試著輸入pip install fairseq --no-build-isolation，可惜的是依舊出現同樣的錯誤，後續上網查到的都是重裝sdk及vs，多次嘗試後依舊沒有用，真的令人頭疼。
[C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.36.32532\include\yvals.h(17): fatal error C1083: 無法開?包含?案: 'crtdbg.h': No such file or directory
      error: command 'C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Tools\\MSVC\\14.36.32532\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2]([url](url))
